src/renderer/components/Input/InputDelegate.vue

```diff
@@ -0,0 +1,362 @@
+import Cycled from 'cycled'
+import InputField from './InputField'
+import truncate from '@/filters/truncate'
+import _ from 'lodash'
+
+export default {
+  name: 'InputDelegate',
+
+  components: {
+    ButtonModal,
+    InputField,
+    ModalQrCodeScanner,
+    MenuDropdown
+  },
+
+  props: {
+    label: {
+      type: String,
+      required: false,
+      default () {
+        return this.$t('SEARCH.DELEGATE')
+      }
+    },
+    name: {
+      type: String,
+      required: false,
+      default: 'delegate'
+    },
+    helperText: {
+      type: String,
+      required: false,
+      default: () => ''
+    },
+    value: {
+      type: String,
+      required: true,
+      default: () => ''
+    },
+    isInvalid: {
+      type: Boolean,
+      required: false,
+      default: false
+    }
+  },
+
+  data: vm => ({
+    inputValue: vm.value,
+    dropdownValue: null,
+    isFocused: false
+  }),
+
+  computed: {
+    currentProfile () {
+      return this.session_profile
+    },
+
+    delegates () {
+      return this.$store.getters['delegate/bySessionNetwork']
+    },
+
+    error () {
+      if (this.$v.model.$dirty && (!this.hasSuggestions || !this.$refs.dropdown.isOpen)) {
+        if (!this.$v.model.required) {
+          return this.$t('INPUT_DELEGATE.ERROR.REQUIRED')
+        } else if (!this.$v.model.isValid) {
+          if (this.inputValue.length <= 20) {
+            return this.$t('INPUT_DELEGATE.ERROR.USERNAME_NOT_FOUND', [this.inputValue])
+          } else if (this.inputValue.length <= 34) {
+            return this.$t('INPUT_DELEGATE.ERROR.ADDRESS_NOT_FOUND', [this.wallet_truncate(this.inputValue)])
+          } else {
+            return this.$t('INPUT_DELEGATE.ERROR.PUBLIC_KEY_NOT_FOUND', [this.wallet_truncate(this.inputValue)])
+          }
+        } else {
+          this.$emit('valid', true)
+        }
+      } else {
+        this.$emit('valid', false)
+      }
+
+      return null
+    },
+
+    hasSuggestions () {
+      return !_.isEmpty(this.suggestions)
+    },
+
+    invalid () {
+      return this.$v.model.$dirty && (this.isInvalid || !!this.error)
+    },
+
+    model: {
+      get () {
+        return this.dropdownValue || this.inputValue
+      },
+      set (value) {
+        this.updateInputValue(value)
+        this.$emit('input', value)
+      }
+    },
+
+    suggestions () {
+      if (!this.currentProfile) return []
+
+      const delegates = _.map(this.delegates, (object) => {
+        const delegate = {
+          name: null,
+          username: object.username,
+          address: object.address,
+          publicKey: object.publicKey
+        }
+
+        delegate.name = `${truncate(object.username, 25)} (${this.wallet_truncate(object.address)})`
+
+        return delegate
+      })
+
+      const results = _.orderBy(delegates, (object) => {
+        return object.name || object.address.toLowerCase()
+      })
+
+      return results.reduce((map, delegate, index) => {
```

Comment:
I'd use a different name for `map`; something like `acc`, `all`, `final`, `delegates`, etc.